File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/client/browsing/library/items/media/files/access/parameters/specs/GivenATypicalItem/WhenGettingFileListParameters.java
package com.lasthopesoftware.bluewater.client.browsing.library.items.media.files.access.parameters.specs.GivenATypicalItem;

import com.lasthopesoftware.bluewater.client.browsing.library.items.Item;
import com.lasthopesoftware.bluewater.client.browsing.library.items.media.files.access.parameters.FileListParameters;

import org.junit.BeforeClass;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class WhenGettingFileListParameters {

	private static final String[] expectedFileListParameters = {
		"Browse/Files",
		"ID=48",
		"Version=2"
	};

	private static String[] returnedFileListParameters;

	@BeforeClass
	public static void before() {
		final FileListParameters fileListParameters = FileListParameters.getInstance();
		returnedFileListParameters = fileListParameters.getFileListParameters(new Item(48));
	}

	@Test
	public void thenTheFileListParametersAreCorrect() {
		assertThat(returnedFileListParameters).containsOnly(expectedFileListParameters);
	}
}

File: projectBlueWater/src/test/java/com/lasthopesoftware/bluewater/client/browsing/files/access/parameters/GivenATypicalItem/WhenGettingFileListParameters.kt
package com.lasthopesoftware.bluewater.client.browsing.files.access.parameters.GivenATypicalItem

import com.lasthopesoftware.bluewater.client.browsing.files.access.parameters.FileListParameters
import com.lasthopesoftware.bluewater.client.browsing.items.ItemId
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class WhenGettingFileListParameters {
    companion object {
        private val expectedFileListParameters = arrayOf(
            "Browse/Files",
            "ID=48",
            "Version=2"
        )
        private val returnedFileListParameters by lazy { FileListParameters.getFileListParameters(ItemId(48)) }
    }

	@Test
	fun thenTheFileListParametersAreCorrect() {
		assertThat(returnedFileListParameters).containsOnly(*expectedFileListParameters)
	}
}
